src/codegen/codegen_cpp_visitor.cpp

```diff
@@ -389,7 +389,7 @@ bool CodegenCVisitor::nrn_state_required() const noexcept {
     if (info.artificial_cell) {
         return false;
     }
-    return info.nrn_state_block != nullptr || breakpoint_exist();
+    return info.nrn_state_block != nullptr || info.breakpoint_with_solve || info.currents.empty();
```

Comment:
Just wanted to verify if `info.currents.empty()` is the same as `currents->next == currents` in MOD2C.